core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -97,9 +97,13 @@ abstract class RawModule extends BaseModule {
   }
 
   private[chisel3] def withRegion[A](newRegion: VectorBuilder[Command])(thunk: => A): A = {
+    _currentRegion ++= stagedSecretCommands
+    stagedSecretCommands.clear()
```

Comment:
An alternative to flushing here would be save/restore the old staged commands. I can't think of a good reason as to why this would be better.

Reply:
Agreed, noted this in the original commit message actually haha because I thought this as well but (also) wasn't sure which was better to be honest.